### PR TITLE
Using Object.is implementation when compare values inside React.PropTypes.oneOf

### PR DESCRIPTION
**Pull request by chicoxyzzy**
implementation of https://github.com/facebook/react/issues/6131

### CODE REVIEW

File: src/isomorphic/classic/types/ReactPropTypes.js [hunk 1]
```diff
@@ -85,6 +85,21 @@ var ReactPropTypes = {
...
+ * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is
+ */
+function is(x, y) {
+  // SameValue algorithm
+  if (x === y) { // Steps 1-5, 7-10
+    // Steps 6.b-6.e: +0 != -0
+    return x !== 0 || 1 / x === 1 / y;
+  } else {
+    // Step 6.a: NaN == NaN
+    return x !== x && y !== y;
```
Comment (chicoxyzzy) [inline src/isomorphic/classic/types/ReactPropTypes.js]: eslint emits warnings here
```
  99:12  warning  Comparing to itself is potentially pointless  no-self-compare
  99:23  warning  Comparing to itself is potentially pointless  no-self-compare

✖ 2 problems (0 errors, 2 warnings)
```

what should I do here?

Comment (zpao) [inline src/isomorphic/classic/types/ReactPropTypes.js]: Just disable that warning for the file.

File: src/isomorphic/classic/types/ReactPropTypes.js [hunk 2]
```diff
@@ -85,6 +85,21 @@ var ReactPropTypes = {
   shape: createShapeTypeChecker,
 };
 
+/**
+ * inlined Object.is polyfill to avoid requiring consumers ship their own
+ * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is
+ */
+function is(x, y) {
```
Comment (syranide) [inline src/isomorphic/classic/types/ReactPropTypes.js]: This should move to a separate file at https://github.com/facebook/react/tree/master/src/shared/stubs

Comment (zpao) [inline src/isomorphic/classic/types/ReactPropTypes.js]: Nah, just going to do this for now.

File: src/isomorphic/classic/types/ReactPropTypes.js (hunk 1 above)
Comment (chicoxyzzy) [inline src/isomorphic/classic/types/ReactPropTypes.js]: done and rebased

File: src/isomorphic/classic/types/ReactPropTypes.js [hunk 3]
```diff
@@ -85,6 +85,23 @@ var ReactPropTypes = {
   shape: createShapeTypeChecker,
 };
 
+/**
+ * inlined Object.is polyfill to avoid requiring consumers ship their own
+ * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is
+ */
+/*eslint-disable no-self-compare*/
+function is(x, y) {
```
Comment (jimfb) [inline src/isomorphic/classic/types/ReactPropTypes.js]: This seems like a weird place to define a polyfill.  Seems like we should do something more analogous to https://github.com/facebook/react/blob/master/src/shared/stubs/Object.assign.js

Comment (zpao) [inline src/isomorphic/classic/types/ReactPropTypes.js]: This is intentional and requested by me due to conversations had with @sebmarkbage a couple months ago, don't worry about it. We can deal with figuring out the polyfill story later.

**Comment by zpao** [comment]
Thanks!

**Pull request merged**